src/file-upload/file-list.tsx

```diff
@@ -175,29 +189,29 @@ export const FileList = ({
         const newRenderModes: FileItemRenderModes = {};
 
         for (const item of fileItems) {
-            /**
-             * We retain the cancelled edits to prevent the edit
-             * display from re-rendering
-             */
-            if (
-                renderModes[item.id] &&
-                renderModes[item.id] === "cancelled-edit"
-            ) {
-                newRenderModes[item.id] = renderModes[item.id];
-            } else if (item.errorMessage) {
+            if (item.errorMessage) {
                 newRenderModes[item.id] = "error";
-            } else if (item.progress < 1) {
-                newRenderModes[item.id] = "loading";
-            } else {
+            } else if (!renderModes[item.id]) {
```

Comment:
does `shouldRenderEditMode` need to check for empty description now? otherwise the `onItemDelete` behaviour for new items only kicks in when the user explicitly clicks edit+cancel, which feels confusing

Reply:
As discussed, added check for undefined and empty string. Also prevent empty string from being able to be saved